extensions/wikia/LinkSuggest/LinkSuggest.class.php

```diff
@@ -364,13 +369,13 @@ static private function replaceResultIfRedirected(&$results, &$redirects) {
 	 * @param $redirects
 	 * @param $results
 	 * @param $exactMatchRow
+	 * @param $limit
 	 *
 	 * @author dymsza
 	 */
 
-	static private function formatResults($db, $res, $query, &$redirects, &$results, &$exactMatchRow) {
-		global $wgLinkSuggestLimit;
-		while(($row = $db->fetchObject($res)) && count($results) < $wgLinkSuggestLimit ) {
+	static private function formatResults($db, $res, $query, &$redirects, &$results, &$exactMatchRow, $limit) {
```

Comment:
Why do we need to add `$limit` here? That should have been handled in the query, right?

Reply:
Well, the global was actually used in the line above and i just passed the limit param to make sure we used the new limit. 

If we'd leave it like it was before the while loop condition would not be met for all the numbers greater than $wgLinkSuggestLimit, so when the query param is bigger than 6 this loop would not execute for numbers bigger than 6.

Am I right though? ;>

Reply:
Initially, yes, b/c I saw that the function signature changed so I had assumed that `count($results) < $limit ` logic was new, but now I see the old implementation is doing the same thing but was using the global. So disregard my comment :)

Unrelated, looks like we probably need to updated [this call](https://github.com/Wikia/app/pull/16774/files/b184b21848d3402726327580163b8e94b42f52a4#diff-2ac62cf91f10b9c15539d927c47d7ec7R231) with the `$limit` parameter as well.

Reply:
Hmmm, the linked line does not contain any calls that need a limit :)

Reply:
Sorry, I meant to link this other `self::formatResults` call on L222... https://github.com/Wikia/app/pull/16774/files/b184b21848d3402726327580163b8e94b42f52a4#diff-2ac62cf91f10b9c15539d927c47d7ec7R222

Reply:
Ah, good catch!